bevy_console_derive/src/lib.rs

```diff
@@ -1,251 +1,51 @@
-        quote! {
-            Some(#description.to_string())
-        }
-    } else {
-        quote! {
-            None
-        }
-    };
-    let command_arg_info = named_fields.iter().map(
-        |syn::Field {
-             attrs, ident, ty, ..
-         }| {
-            let name = ident.as_ref().unwrap().to_string();
-            let ty_string = ty_to_string(ty)
-                .map(|ty_string| quote!(#ty_string))
-                .unwrap_or_else(|| quote!(stringify!(#ty)));
-            let doc_comments = get_doc_comments(attrs);
-            let arg_description = if !doc_comments.is_empty() {
-                let description = doc_comments.join("\n");
-                quote! {
-                    Some(#description.to_string())
-                }
-            } else {
-                quote! {
-                    None
-                }
-            };
-            let optional = is_ty_option(ty);
-
-            quote! {
-                bevy_console::CommandArgInfo {
-                    name: #name.to_string(),
-                    ty: #ty_string.to_string(),
-                    description: #arg_description,
-                    optional: #optional,
-                }
-            }
-        },
-    );
+    let name_string = get_command_name(&derive_input);
+    let name = &derive_input.ident;
+    let generics = derive_input.generics;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
 
     TokenStream::from(quote! {
-        #[automatically_derived]
-        impl bevy_console::CommandName for #ident {
-            fn command_name() -> &'static str {
-                #command_name
+        impl #impl_generics bevy_console::NamedCommand for #name #ty_generics #where_clause {
+            fn name() -> &'static str {
+                #name_string
             }
         }
 
-        #[automatically_derived]
-        impl bevy_console::CommandArgs for #ident {
-            fn from_values(values: &[bevy_console::ValueRawOwned]) -> ::std::result::Result<Self, bevy_console::FromValueError> {
-                let mut values = values.iter();
-
-                Ok(#ident {
-                    #( #fields )*
-                })
-            }
-        }
-
-        #[automatically_derived]
-        impl bevy_console::CommandHelp for #ident {
-            fn command_help() -> ::std::option::Option<bevy_console::CommandInfo> {
-                ::std::option::Option::Some(bevy_console::CommandInfo {
-                    name: #command_name.to_string(),
-                    description: #command_description,
-                    args: vec![
-                        #( #command_arg_info, )*
-                    ],
-                })
-            }
-        }
-
-        #[automatically_derived]
-        impl bevy::prelude::Resource for #ident {}
+        impl #impl_generics bevy::prelude::Resource for #name #ty_generics #where_clause {};
     })
 }
 
-fn get_doc_comments(attrs: &[syn::Attribute]) -> Vec<String> {
```

Comment:
Did we lose this functionality, or is it handled elsewhere?

Reply:
Clap should convert documentation on fields to argument hints/help strings etc automatically